src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -235,10 +244,23 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
             response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
             response.Capabilities.Add("TypedDataCollection", bool.TrueString);
+            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
 
             return response;
         }
 
+        internal void WorkerTerminateRequestHandler(WorkerTerminate request)
+        {
+            try
+            {
+                _hostApplicationLifetime.StopApplication();
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("Error during terminating the worker process", ex);
```

Comment:
What code path will catch this exception? I do not think we have a try-catch on the caller of this method.  I think we should just remove the try-catch here.

Reply:
done